src/jdk.incubator.vector/share/classes/jdk/incubator/vector/ByteVector.java

```diff
@@ -1885,21 +1885,21 @@ M testTemplate(Class<M> maskType, Test op) {
     @ForceInline
     final
     <M extends VectorMask<Byte>>
-    M compareTemplate(Class<M> maskType, Comparison op, Vector<Byte> v) {
+    M compareTemplate(Class<M> maskType, Comparison op, Vector<Byte> v, M m) {
         Objects.requireNonNull(v);
```

Comment:
It's better to check the argument `M m` for masked API, since we'd better to make sure it is not "null" when going into hotspot intrinsics. Calling `m.check(maskType, this)` can guarantee this.

Reply:
Yes, please follow the pattern for the other templates, where we move the default Java implementation to a method that is shared by both templates. It helps to be consistent and also uniformly apply the mask checking only where needed (we know that works).
Perhaps it might be possible to uniformly collapse the pattern later as we push more mask cases into the JIT.